### PR TITLE
Enhance tab behaviour in drawer and choice editor

### LINKED ISSUE #18095: Tab order in Choice editor goes outside page after value
### Describe the Bug

When adding many choices for a Dropdown type element, you're mostly using the keyboard as you're typing a bunch of fields, so it would make sense to confirm the entry with the keyboard, too. However, when you press tab after being in the Value field, the focus is brought to the browser search bar.

You _can_ get to the accept button by Shift+Tabbing twice to go back prior to the Text field, but there's a couple problems with that
* It's misleading since you have to go back to complete the workflow
* The visual indication on the checkmark is very subtle, so it is nearly impossible to tell you've focused correctly.
* If you shift-tab too far, focus actually goes back to the things underneath the popup sidebar, which is also bad

### To Reproduce

* Go to the data model editor for a collection
* Add a Dropdown type item
* Click the button to add a choice, and then try to navigate from there with the keyboard

### Hosting Strategy

Self-Hosted (Docker Image)

### PR DESCRIPTION
**Pull request by gitstart**
Fixes #18095 

## Demo
https://www.loom.com/share/01690fe6d4a145c39d27c584dd015a0f?sid=18ea89a3-e85c-456a-8b5f-7bedde1d406c

---
This code was written and reviewed by GitStart Community. Growing great engineers, one PR at a time.

### CODE REVIEW

File: .changeset/mighty-keys-agree.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@directus/app': minor
+---
+
+- Add support for the confirmation of entry of choices into Dropdown component with `Tab` key
+- Also, prevent the focus from leaving the sidebar when pressing `Shift + Tab`
```
Comment (paescuj) [inline .changeset/mighty-keys-agree.md:6]:
```suggestion
Enhanced UX in dropdown choice editor by leading the focus to the save button after the last field when using `Tab` key as well as preventing the focus from leaving the sidebar when pressing `Shift + Tab`
```

**Review by paescuj — CHANGES_REQUESTED** [review]
We should improve the button focus style directly in the `v-button` component. 
This ensures the focus on cancel button is also visible (not the case right now).

**Comment by rijkvanzanten** [comment]
Relying on the keydown event listener with the `tab` key specifically doesn't feel like the correct approach to me. I'm hopeful we can find a solution that relies on `tabindex` and "regular" browser native tab handling for the experience we're looking for 👍🏻 

I'll close this for now, as my proposed direction is a totally different approach to solving the same problem 🙂 Thanks anyways! 🚀

**Comment by rijkvanzanten** [comment]
A focus trap around a drawer etc makes a ton of sense, but that's not something that should be implemented through event handlers on the individual inputs themselves, but rather on a component context wrapper 👍🏻

**Pull request closed without merging**